File: 1500-count-largest-group/count-largest-group.java
class Solution {
    public int countLargestGroup(int n) {
        int max = 0;
        int ans = 0;
        int[] map = new int[40];
        for(int i = 1;i<=n;i++){
            int key = 0,i0=i;
            while(i0!=0){
                key+=(i0%10);
                i0/=10;
            }
            map[key]++;
            if(max<map[key]){
                max=map[key];
                ans=1;
            }else if(map[key]==max){
                ans++;
            }
        }
        return ans;
    }
}

File: 1500-count-largest-group/count-largest-group.cpp
class Solution {
public:
    int countLargestGroup(int n) {
        int map[40] = {0};
        int maxGroup = 0, ans = 0;

        for (int i = 1; i <= n; ++i) {
            int sum = 0, num = i;
            while (num > 0) {
                sum += num % 10;
                num /= 10;
            }
            map[sum]++;
            if (map[sum] > maxGroup) {
                maxGroup = map[sum];
                ans = 1;
            } else if (map[sum] == maxGroup) {
                ans++;
            }
        }

        return ans;
    }
};